imap_data_access/io.py

```diff
@@ -159,6 +163,85 @@ def query(
             "At least one query parameter must be provided. "
             "Run 'query -h' for more information."
         )
+    # Check instrument name
+    if instrument is not None:
+        if instrument not in (
+            "codice",
+            "glows",
+            "hi",
+            "hit",
+            "idex",
+            "lo",
+            "mag",
+            "swapi",
+            "swe",
+            "ultra",
+        ):
+            raise ValueError(
+                "Not a valid instrument, please choose from "
+                "('codice', 'glows', 'hi', 'hit', 'idex', "
+                "'lo', 'mag', 'swapi', 'swe', 'ultra')"
+            )
+
+    # Check data-level
+    # do an if statement that checks that data_level was passed in,
+    # then check it against all options, l0, l1a, l1b, l2, l3 etc.
+    if data_level is not None:
+        if data_level not in (
```

Comment:
Ditto for valid data levels. I actually need to remove L2pre, so it would be good to put it in one place